### PR TITLE
feat(layout): rename SegmentSize to Spread

### PR DESCRIPTION
**Pull request by joshka**
For some time we've been using the term "segment size" to refer to the
way extra space is distributed when the constraints are satisfied. This
is a very confusing name, and it's not clear what it means. This commit
renames it to "spread", which is a more common term for this concept.

Removed the unstable feature gate for this API now that a good name has
been found.

BREAKING CHANGE:
- SegmentSize has been renamed to Spread
- unstable-segment-size feature has been removed
- Layout::segment_size() has been renamed to Layout::spread()
- Table::segment_size() has been renamed to Table::spread()

### CODE REVIEW

**Comment by joshka** [comment]
I want a couple of approvals on this so we have consensus on the name.

**Comment by kdheepak** [comment]
Is it a breaking change if this was gated behind a unstable flag?

**Comment by joshka** [comment]
> Is it a breaking change if this was gated behind a unstable flag?

Yeah, but not one I'd worry about too much. Doesn't hurt to mark it as such.

**Comment by kdheepak** [comment]
The title probably needs a `!` then, right? 

For some reason the `check-breaking-change-label` action isn't catching it.

**Review by Valentin271 — APPROVED** [review]
I like this name a lot more.

> For some reason the `check-breaking-change-label` action isn't catching it.

If I'm not mistaken the action only checks for `!` in the title and, if found, adds the `breaking-change` label.

**Review by orhun — APPROVED** [review]
`Spread` stills does not sound like the exact definition of this - or maybe it is because I got used to `SegmentSize` at this point. I will give it some time to grow on me.

**Comment by kdheepak** [comment]
Can we hold off on merging this? Or still keep it behind the unstable flag? 

I like the name `Spread` as a verb describing what we are doing with the remaining segment size, but the variant names don't quite make sense now, right?

**Comment by orhun** [comment]
Agree with kd here!

**Comment by joshka** [comment]
The enum describes an action that happens (the allocation of remaining space once all constraints are satisfied). I think this would suggest a verb name is more appropriate than a noun for that concept. SegmentSize has never been clear to me. What is a segment? Why is being sized?

If you do feel more nounish, then something that notifies a verb is good. Eg allocate extra space -> ExtraSpaceAllocation. Distribute-> distribution ("spread" -> "spread" as it is bough a verb and the result of that verb having been done - this seems convenient).

I like the variant renames suggested.

**Comment by joshka** [comment]
#804 makes this no longer necessary.

**Pull request closed without merging**
